apis/v1beta1/vspherefailuredomain_webhook.go

```diff
@@ -90,4 +90,15 @@ func (r *VSphereFailureDomain) Default() {
 	if r.Spec.Region.AutoConfigure == nil {
 		r.Spec.Region.AutoConfigure = pointer.Bool(false)
 	}
+
+	// Converts the old Networks field to NetworkConfigs
+	// TODO (@rkatz) - Can this be harmful for Gitops and other users that check spec and re-apply? Probably it will generate
+	// a difference between "what is expected" and "what we have" as there's going to be the additional "NetworkConfigs" field
+	//
+	if len(r.Spec.Topology.NetworkConfigs) == 0 && len(r.Spec.Topology.Networks) > 0 {
+		r.Spec.Topology.NetworkConfigs = make([]FailureDomainNetwork, len(r.Spec.Topology.Networks))
+		for i := range r.Spec.Topology.Networks {
+			r.Spec.Topology.NetworkConfigs[i].NetworkName = r.Spec.Topology.Networks[i]
+		}
+	}
```

Comment:
I'm still ramping up on CAPV, but I think we should not convert this in the webhook to the other result.

IMHO we should:
* block if both are set (`NetworkConfigs` and `Networks`)
* Handle converting to `NetworkConfigs` internally in the controller to support both settings for now
* Handle migration to the new field, when introducing a new api version (via conversion webhook)

Reply:
Sound good. I have added the conversion internally as well, but wanted to catch it early. Will change it here

Reply:
ok, dropped mutation, moved the validation to validate creation.
Internally the controller already does this conversion as well, as you pointed below :)